server/src/main/java/org/opensearch/index/mapper/TextFieldMapper.java

```diff
@@ -449,7 +449,7 @@ private PrefixFieldMapper buildPrefixMapper(BuilderContext context, FieldType fi
                 pft.setStoreTermVectorOffsets(true);
             }
             PrefixFieldType prefixFieldType = new PrefixFieldType(tft, fullName + "._index_prefix", indexPrefixes.get());
-            prefixFieldType.setAnalyzer(analyzers.getIndexAnalyzer());
+            prefixFieldType.setAnalyzer(tft.indexAnalyzer());
```

Comment:
For consistency with `PhraseFieldType`, can you please move the call to `prefixFieldType.setAnalyzer` into the `PrefixFieldType` constructor?

Reply:
That makes sense, I've done that.